src/Adapter/Order/CommandHandler/UpdateProductInOrderHandler.php

```diff
@@ -0,0 +1,247 @@
+        }
+
+        $old_quantity = $orderDetail->product_quantity;
+
+        $orderDetail->product_quantity = $product_quantity;
+        $orderDetail->reduction_percent = 0;
+
+        // update taxes
+        $res &= $orderDetail->updateTaxAmount($order);
+
+        // Save order detail
+        $res &= $orderDetail->update();
+
+        // Update weight SUM
+        $order_carrier = new OrderCarrier((int) $order->getIdOrderCarrier());
+        if (Validate::isLoadedObject($order_carrier)) {
+            $order_carrier->weight = (float) $order->getTotalWeight();
+            $res &= $order_carrier->update();
+            if ($res) {
+                $order->weight = sprintf('%.3f ' . Configuration::get('PS_WEIGHT_UNIT'), $order_carrier->weight);
+            }
+        }
+
+        // Save order invoice
+        if (isset($orderInvoice)) {
+            $res &= $orderInvoice->update();
+        }
+
+        // Update product available quantity
+        StockAvailable::updateQuantity(
+            $orderDetail->product_id,
+            $orderDetail->product_attribute_id,
+            $old_quantity - $orderDetail->product_quantity,
+            $order->id_shop
+        );
+
+        $order = $order->refreshShippingCost();
+
+        if (!$res) {
+            throw new OrderException('An error occurred while editing the product line.');
+        }
+
+        Hook::exec('actionOrderEdited', ['order' => $order]);
+    }
+
+    /**
+     * @param UpdateProductInOrderCommand $command
+     * @param OrderDetail $orderDetail
+     * @param Order $order
+     * @param OrderInvoice|null $orderInvoice
+     *
+     * @throws OrderException
+     */
+    private function assertProductCanBeUpdated(
+        UpdateProductInOrderCommand $command,
+        OrderDetail $orderDetail,
+        Order $order,
+        OrderInvoice $orderInvoice = null
+    ) {
+        if (!Validate::isLoadedObject($orderDetail)) {
+            throw new OrderException('The Order Detail object could not be loaded.');
+        }
+
+        if (null !== $orderInvoice && !Validate::isLoadedObject($orderInvoice)) {
+            throw new OrderException('The invoice object cannot be loaded.');
+        }
+
+        if (!Validate::isLoadedObject($order)) {
+            throw new OrderException('The order object cannot be loaded.');
+        }
+
+        if ($orderDetail->id_order != $order->id) {
+            throw new OrderException('You cannot edit the order detail for this order.');
+        }
+
+        // We can't edit a delivered order
+        if ($order->hasBeenDelivered()) {
+            throw new OrderException('You cannot edit a delivered order.');
+        }
+
+        if (null !== $orderInvoice && $orderInvoice->id_order != $order->id) {
+            throw new OrderException('You cannot use this invoice for the order');
+        }
+
+        // Clean price
+        $product_price_tax_incl = str_replace(',', '.', $command->getPriceTaxIncluded());
```

Comment:
I think no, because that would mean fixing issue quietly (as it is now here). I'll add todo for it so we can improve it, but what I think is that we should use Decimal or maybe Price VO that has validation inside.